src/PostListingDisplay/components/PostListingDisplay.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+
+import InputField from './InputField';
+
+import '../styles/PostListingDisplay.css';
+
+/**Component class that displays the input fields for a user to post a listing */
+class PostListingDisplay extends Component {
+    constructor(props){
+        super(props);
+
+        this.state = {
+            title: '',
+            description: '',
+            numRooms: '',
+            numSingles: '',
+            singlePrice: '',
+            numShared: '',
+            sharedPrice: '',
+            numBathrooms: '',
+            leaseTypes: '',
+            startDate: '',
+            endDate: '',
+            listingPrice: ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({ [name]: value }, () => {
+            if(name === "numSingles" || name === "singlePrice" || name === "numShared" || name === "sharedPrice"){
```

Comment:
Rather than manually checking which fields are being updated, is there a reason not to use a dedicated handler for mutations to those fields specifically?

Reply:
I hadn't thought of that, but that is much better. I added a function called handlePriceChange() that calls handlechange and calculateListingPrice and changed the onchange function for the 4 price determining inputs to handlePriceChange()